### ISSUE TITLE
capi docs: hyper_buf_copy() can return NULL, right?

### ISSUE THREAD

**Issue by bagder** [hyperium/hyper#2563]
The current comment for `hyper_buf_copy` in the header says:
~~~c
/*
 Create a new `hyper_buf *` by copying the provided bytes.

 This makes an owned copy of the bytes, so the `buf` argument can be
 freed or changed afterwards.
*/
~~~

I think it would be helpful to users to mention that this can return NULL in cases of out of memory. Because it *can* return that then, can't it?

**Comment by seanmonstar** [comment]
Yes, you're right! This would be a simple doc change. And in fact the code is already prepared to return NULL on panic. However, it seems there's no progress on the issue to allow setting OOM to panic instead of abort (I recorded my check-in in #2396).
